extensions/vscode-aml/src/web/extension.ts

```
@@ -0,0 +1,267 @@
+
+function openInAzimuttUrl(aml: string): string {
+	return 'https://azimutt.app/create?aml=' + encodeURIComponent(aml)
+}
+
+// see https://microsoft.github.io/monaco-editor/typedoc/interfaces/languages.DocumentSymbolProvider.html
+class AmlDocumentSymbolProvider implements DocumentSymbolProvider {
+	provideDocumentSymbols(document: TextDocument, token: CancellationToken): ProviderResult<SymbolInformation[] | DocumentSymbol[]> {
+		const symbols: DocumentSymbol[] = []
+		const regex = /(^|\n)(type\s+)?((?:[a-zA-Z_][a-zA-Z0-9_]*\.)?[a-zA-Z_][a-zA-Z0-9_]*)/g
+		let match: RegExpExecArray | null = null
+		while (match = regex.exec(document.getText())) {
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)